File: src/main/java/uk/laxd/dndSimulator/action/SimpleTargetSelector.java
package uk.laxd.dndSimulator.action;

import uk.laxd.dndSimulator.character.Character;

public class SimpleTargetSelector extends TargetSelector {

    private Character target;

    public SimpleTargetSelector(Character target) {
        this.target = target;
    }

    @Override
    public Character getTarget() {
        return target;
    }
}

File: app/src/main/kotlin/uk/laxd/dndSimulator/action/SimpleTargetSelector.kt
package uk.laxd.dndSimulator.action

import uk.laxd.dndSimulator.character.Character

class SimpleTargetSelector(
    private val characters: Collection<Character>
) : TargetSelector {
    override fun findTarget(actor: Character): Character? {
        // Find someone on a team that isn't the Characters team!
        return characters.filter { c -> c.hp > 0 }
            .firstOrNull { c -> c.team != actor.team }
    }
}